### PR TITLE
 SelectBox - Add valueExpr in the Search and editing demos 

### PR DESCRIPTION
**Pull request by artem-kurchenko**

### CODE REVIEW

File: MVCDemos/Views/SelectBox/SearchAndEditing.cshtml [hunk 1]
```diff
@@ -113,16 +114,12 @@
...
-        Name: "HD Video Player",
-        ID: 0
-    }
-
     function selectBox_valueChanged(data) {
         var $result = $(".current-value");
 
-        if(data.value) {
-            $result.text(data.value.Name + " (ID: " + data.value.ID + ")");
+        if (data.value!== null) {
```
Comment (DokaRus) [inline MVCDemos/Views/SelectBox/SearchAndEditing.cshtml]:
```suggestion
        if (data.value !== null) {
```

File: NetCoreDemos/Views/SelectBox/SearchAndEditing.cshtml [hunk 2]
```diff
@@ -115,16 +116,12 @@
...
-        Name: "HD Video Player",
-        ID: 0
-    }
-
     function selectBox_valueChanged(data) {
         var $result = $(".current-value");
 
-        if(data.value) {
-            $result.text(data.value.Name + " (ID: " + data.value.ID + ")");
+        if (data.value !==null) {
```
Comment (DokaRus) [inline NetCoreDemos/Views/SelectBox/SearchAndEditing.cshtml]:
```suggestion
        if (data.value !== null) {
```

File: MVCDemos/Views/SelectBox/SearchAndEditing.cshtml (hunk 1 above)
Comment (artem-kurchenko) [reply MVCDemos/Views/SelectBox/SearchAndEditing.cshtml]: done

File: NetCoreDemos/Views/SelectBox/SearchAndEditing.cshtml (hunk 2 above)
Comment (artem-kurchenko) [reply NetCoreDemos/Views/SelectBox/SearchAndEditing.cshtml]: done

File: JSDemos/Demos/SelectBox/SearchAndEditing/Angular/app/app.component.html [hunk 3]
```diff
@@ -24,15 +24,15 @@
                     <dx-select-box #selectBox
                         [dataSource]="productsDataSource"
                         [acceptCustomValue]="true"
-                        displayExpr="Name"
+                        displayExpr="Name"
```
Comment (onebalaban) [inline JSDemos/Demos/SelectBox/SearchAndEditing/Angular/app/app.component.html]: unnecessary spaces at the end of the line

File: JSDemos/Demos/SelectBox/SearchAndEditing/Angular/app/app.component.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ export class AppComponent {
     products: Product[];
     simpleProducts: SimpleProduct[];
     productsDataSource: DataSource;
-    product: number;
+    product: number;
```
Comment (onebalaban) [inline JSDemos/Demos/SelectBox/SearchAndEditing/Angular/app/app.component.ts]: unnecessary spaces at the end of the line

File: JSDemos/Demos/SelectBox/SearchAndEditing/Angular/app/app.component.ts [hunk 5]
```diff
@@ -38,7 +38,7 @@ export class AppComponent {
     constructor(service: Service) {
         this.products = service.getProducts();
         this.simpleProducts = service.getSimpleProducts();
-        this.product = this.simpleProducts[0].ID;
+        this.product = this.simpleProducts[0].ID;
```
Comment (onebalaban) [inline JSDemos/Demos/SelectBox/SearchAndEditing/Angular/app/app.component.ts]: unnecessary spaces at the end of the line

**Review by onebalaban — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

**Pull request merged**